deeppavlov/dataset_iterators/morphotagger_iterator.py

```diff
@@ -103,14 +103,15 @@ def gen_batches(self, batch_size: int, data_type: str = 'train',
         if shuffle is None:
             shuffle = self.shuffle
         data = self.data[data_type]
-        if shuffle:
-            random.shuffle(data)
         lengths = [len(x[0]) for x in data]
         indexes = np.argsort(lengths)
         L = len(data)
         if batch_size < 0:
             batch_size = L
-        for start in range(0, L, batch_size):
+        starts = list(range(0, L, batch_size))
+        if shuffle:
+            np.random.shuffle(starts)
```

Comment:
```suggestion
            self.random.shuffle(starts)
```
in the `split()` method as well

Reply:
Done.